### PR TITLE
nodejs-warmup-week2

### PR DESCRIPTION
**Pull request by viedamayanti**

### CODE REVIEW

File: nodejs/week2/app.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+const express = require('express');
+const app = express();
+const port = process.env.PORT || 3000;
+const doc = require('./documents.json');
+// const knex = require("./database")
```
Comment (orhantoy) [inline nodejs/week2/app.js:5]: I would remove this line.

File: nodejs/week2/app.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+});
+
+app.get('/documents/:id', (req, res) => {
+  const filtered = doc.some(
+    (document) => document.id === parseInt(req.params.id)
+  );
+  if (filtered) {
+    res.json(doc.filter((document) => document.id === parseInt(req.params.id)));
+  } else {
+    res.status(404).send(`The document ${req.params.id} is not found`);
```
Comment (orhantoy) [inline nodejs/week2/app.js:43]: This responds with text instead of JSON.

File: nodejs/week2/app.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+});
+
+// It does not work
+app.get('/search', (req, res) => {
+  console.log(req.query);
+  const queryValue = req.query.q;
+  if (queryValue) {
+    const values = doc.filter((item) => {
+      return Object.values(item)
+        .map(String)
```
Comment (orhantoy) [inline nodejs/week2/app.js:20]: Nice trick 👍

File: nodejs/week2/app.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+const doc = require('./documents.json');
+// const knex = require("./database")
+// Support parsing JSON requests
+app.use(express.json()); //to accept json request
+
+app.get('/', (req, res) => {
+  res.send('This is a search engine');
+});
+
+// It does not work
```
Comment (orhantoy) [inline nodejs/week2/app.js:13]: It seems to work? 🤔 

```
$ curl localhost:3000/search
[{"id":1,"name":"Used Apple Airpods","price":"50","description":"Battery life is not great","type":"pdf","value":"Hello world"},{"id":2,"name":"Used Apple MacBook","price":"35000","description":"Battery life is not great","type":"electronic","value":"Mac"},{"id":3,"name":"Used iPhone","price":"3500","description":"Battery life is not great","type":"mobile","value":"The phone"}]

$ curl localhost:3000/search?q=mac
[{"id":2,"name":"Used Apple MacBook","price":"35000","description":"Battery life is not great","type":"electronic","value":"Mac"}]
```

File: nodejs/week2/app.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+          value.toLowerCase().includes(queryValue.toLowerCase())
+        );
+    });
+    return res.json(values);
+  }
+  return res.json(doc);
+});
+
+app.post('/search', (req, res) => {
+  console.log(req.body.fields);
```
Comment (orhantoy) [inline nodejs/week2/app.js:32]: Missing the implementation here.

**Review by orhantoy — COMMENTED** [review]
Looks great! I think you're just missing the `POST /search` route.

**Review by Yuliia-Balandiuk — COMMENTED** [review]
Hi Vie. 
You did really good work!

File: nodejs/week2/app.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+// const knex = require("./database")
+// Support parsing JSON requests
+app.use(express.json()); //to accept json request
+
+app.get('/', (req, res) => {
+  res.send('This is a search engine');
+});
+
+// It does not work
+app.get('/search', (req, res) => {
```
Comment (Yuliia-Balandiuk) [inline nodejs/week2/app.js:14]: It doesn't work for me too, I don't understand why

File: nodejs/week2/app.js (hunk 1 above)
Comment (Yuliia-Balandiuk) [inline nodejs/week2/app.js:5]: It is good practice to remove commented code

File: nodejs/week2/app.js (hunk 6 above)
Comment (orhantoy) [reply nodejs/week2/app.js:14]: Do you mean Vie's code does not work? What error(s) are you seeing?

Comment (Yuliia-Balandiuk) [reply nodejs/week2/app.js:14]: It's magic, today it works)))). Yesterday I received HTML in my body, but it was my mistake.
image

File: nodejs/week2/app.js (hunk 4 above)
Comment (viedamayanti) [reply nodejs/week2/app.js:13]: It works 😅

**Comment by viedamayanti** [comment]
Thank you so much for the feedback everyone 🙏

**Pull request merged**